core/src/main/resources/META-INF/services/io.onetable.spi.sync.TargetClient

```diff
@@ -0,0 +1,21 @@
+##########################################################################
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+##########################################################################
+
+io.onetable.delta.DeltaClient
```

Comment:
If a new service implementation is added, does it require a rebuild?

Reply:
Well, it does if you are committing the implementation to onetable. Not if you are just trying to use a new one by adding a custom jar to the classpath where this same file would be added. Each packaged extension would essentially implement the various interface and provide one of these files in META-INF/services/{interface-classname}.

If you are adding to the above file then you are adding the new client to the project and you need a new build anyway.

Reply:
This is great, thanks for the clarification. Just to confirm, consider an e.g., the core package has 3 implementations of `TargetClient` interface through `core\META-INF\services\TargetClient`. If a new package foobar is added to classpath, and feebar contains 2 different implementations in `foobar\META-INF\services\TargetClient`, then the `ServiceLoader` will return total 5 implementation.

Ideally, the core library and the implementations should live in their own packages. Ability to load implementation from multiple packages would make it easier.

Reply:
@ashvina - ServiceLoader already allows this. Assuming the packages are in different jars. If they are not in separate jars then the META-INF will be shared. It does not matter what the implementation package name is on the fully qualified name of the interface that they all implement.

Reply:
Here is a really good article on ServiceLoader that describes the intent and details quite well. It also points to a google annotation project for automating the service config file which I wasn't aware of - pretty nice! https://pedrorijo.com/blog/java-service-loader/ I am going to add this to the related Issue description too.

Reply:
Great, thanks for the pointer. `@AutoService` should make it easier to integrate new implementations.